boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -83,7 +83,7 @@ endif
 CC = $(CROSSDEV)cc
 CXX = $(CROSSDEV)c++
 CPP = $(CROSSDEV)cc -E -P -x c
-LD = $(CROSSDEV)ld
+LD = $(CROSSDEV)gcc
```

Comment:
This is breaking our clang-based build.

Reply:
```patch
diff --git a/boards/sim/sim/sim/scripts/Make.defs b/boards/sim/sim/sim/scripts/Make.defs
index 65600e15ec..e4615f12f1 100644
--- a/boards/sim/sim/sim/scripts/Make.defs
+++ b/boards/sim/sim/sim/scripts/Make.defs
@@ -83,7 +83,7 @@ endif
 CC = $(CROSSDEV)cc
 CXX = $(CROSSDEV)c++
 CPP = $(CROSSDEV)cc -E -P -x c
-LD = $(CROSSDEV)gcc
+LD = $(CROSSDEV)cc
 ifeq ($(CONFIG_HOST_MACOS),y)
 STRIP = $(CROSSDEV)strip
 AR = $(TOPDIR)/tools/macar-rcs.sh
```
@lulingar Please, try applying this patch

Reply:
Thanks, it failed further down the build, like:

    echo "LD:  nuttx"
    LD:  nuttx
    cc -r  -L"/nuttx/staging" -L board  -o nuttx.rel up_head.o --start-group -lsched -ldrivers -lboards -lc -lmm -larch -lapps -lnet -lfs -lbinfmt -lboard  --end-group
    clang: error: unsupported option '--start-group'
    clang: error: unsupported option '--end-group'
    make[1]: Leaving directory '/nuttx/arch/sim/src'
    make[1]: *** [Makefile:314: nuttx] Error 1
    make: *** [tools/Makefile.unix:422: nuttx] Error 2

But it seems #3904 fixes this, I'm testing it out.

Reply:
It didn't fix it, in our clang-only build we don't have gcc. I'm testing some other ways to go about this. But this begs the question, why forcing this way the compiler to be gcc?

Reply:
@lulingar Do you have any instructions how to config clang build? What OS do you use?

Reply:
I've successfully with

In my Ubuntu 21.04 machine, I've just successfully built `sim:nsh` with clang. For forcing the `clang` compiler since it's not symlinked to `cc`, I've applied the following patch:
```patch
diff --git a/boards/sim/sim/sim/scripts/Make.defs b/boards/sim/sim/sim/scripts/Make.defs
index 067914aea2..0d2a861890 100644
--- a/boards/sim/sim/sim/scripts/Make.defs
+++ b/boards/sim/sim/sim/scripts/Make.defs
@@ -80,10 +80,10 @@ ifeq ($(CONFIG_SIM_M32),y)
   ARCHCPUFLAGSXX += -m32
 endif
 
-CC = $(CROSSDEV)cc
-CXX = $(CROSSDEV)c++
-CPP = $(CROSSDEV)cc -E -P -x c
-LD = $(CROSSDEV)gcc
+CC = $(CROSSDEV)clang
+CXX = $(CROSSDEV)clang++
+CPP = $(CROSSDEV)clang -E -P -x c
+LD = $(CROSSDEV)clang
 ifeq ($(CONFIG_HOST_MACOS),y)
 STRIP = $(CROSSDEV)strip
 AR = $(TOPDIR)/tools/macar-rcs.sh
```
Also, I've successfully built `sim:nsh` on **macOS** with clang as the default compiler.

@lulingar, from the command line you've posted we can see that `--start-group` was incorrectly passed instead of `-Wl,--start-group`. Have you performed a `distclean` after you pulled the latest changes from upstream? Since the Makefiles from the board were changed, these are not automatically refreshed by `make` because the NuttX build system makes a copy of them on the target `configure` step.

Reply:
Thank you for the indications. I have tested aliasing clang as gcc, after the merge of #3904, and it worked.

Reply:
@lulingar, I've opened https://github.com/apache/incubator-nuttx/pull/3921 for fixing the support for building the simulator on systems with Clang as the default compiler.